Library/Homebrew/test/utils/svn_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "utils/svn"
+
+describe Utils do
+  describe "#self.svn_available?" do
+    it "returns true if svn --version command succeeds" do
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(true)
+      expect(described_class.svn_available?).to be_truthy
+    end
+
+    it "returns false if svn --version command does not succeed" do
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(false)
+      expect(described_class.svn_available?).to be_falsey
+    end
+
+    it "returns svn version if already set" do
+      described_class.instance_variable_set(:@svn, true)
```

Comment:
This variable will be set only if `svn_available?` has been called successfully atleast once. So, there is no method which can ensure `@svn` will already exist if svn is not available on the system.
I'm removing `@svn` before each test now. I don't think svn_version already set can work without setting `@svn` because on a system without svn, this test won't work the way it should.